### PR TITLE
New Request Scenes

### PR DESCRIPTION
**Pull request by salolivares**
## Main features
- Got rid of the popup dialog
- New request scene features an item list, ability to add items, and a datapicker
- New item scene allows you to add the name of the item and set how much of that item you want.
- Form validation!

### Misc.
- Main request list now updates on launch. No need to press refresh!!! :DD
- Removed more duplicate code in RequestList.js

cc @spencerprescott @brianyan

### CODE REVIEW

File: Dispatchr/app/actions/types.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
 export const ADD_NEW_REQUEST = 'ADD_NEW_REQUEST';
 export const REQUEST_REQUESTED = 'REQUEST_REQUESTED';
 export const REQUEST_RECEIVED = 'REQUEST_RECEIVED';
+export const ADD_ITEM = 'ADD_ITEM';
```
Comment (brianyan) [inline Dispatchr/app/actions/types.js]: Aren't we supposed to just include all these consts in one actions file

File: Dispatchr/app/components/ItemList.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    const rowAction = () => {
+      highlightRow(sectionId, rowId);
+      Actions.EditItemForm({rowData});
+    };
+    return (
+      <TouchableHighlight onPress={rowAction}>
+         <View>
+           <View style={styles.rowContainer}>
+             <Text style={styles.text}>
+               {rowData.itemName} - Qty. {rowData.quantity}
```
Comment (brianyan) [inline Dispatchr/app/components/ItemList.js]: what does Qty. do? I don't see it specified elsewhere

File: Dispatchr/app/components/NewItemForm.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import { Text, TextInput,  View, StyleSheet, TouchableHighlight} from 'react-native';
+import { connect } from 'react-redux';
+import { ActionCreators } from '../actions';
+import { bindActionCreators } from 'redux';
+
```
Comment (brianyan) [inline Dispatchr/app/components/NewItemForm.js]: can you link library documentation here.

File: Dispatchr/app/containers/RequestsList.js [hunk 4]
```diff
@@ -33,22 +32,26 @@ class RequestsList extends Component {
...
+        dataSource: this.state.dataSource.cloneWithRows(nextProps.requests),
+        showFetchButton: nextProps.requests.length == 0
+      })
+    }
   }
+  
 
   _leftSideSelected() {
-    console.log("sfdsaf")
+    console.log("left")
```
Comment (brianyan) [inline Dispatchr/app/containers/RequestsList.js]: do we want this in master?

File: Dispatchr/app/containers/RequestsList.js [hunk 5]
```diff
@@ -33,22 +32,26 @@ class RequestsList extends Component {
...
   }
 
   _rightSideSelected() {
-    console.log("asfd")
+    console.log("right")
```
Comment (brianyan) [inline Dispatchr/app/containers/RequestsList.js]: do we want this in master

File: Dispatchr/app/containers/RequestsList.js (hunk 4 above)
Comment (salolivares) [inline Dispatchr/app/containers/RequestsList.js]: Yea, more descriptive print statements!

File: Dispatchr/app/containers/RequestsList.js (hunk 5 above)
Comment (salolivares) [inline Dispatchr/app/containers/RequestsList.js]: see other comment

File: Dispatchr/app/actions/types.js (hunk 1 above)
Comment (salolivares) [inline Dispatchr/app/actions/types.js]: We are! I just didn't prioritize it right now. If you really want it, I can switch it to one actions file.

Comment (brianyan) [inline Dispatchr/app/actions/types.js]: It's okay.

**Comment by brianyan** [comment]
Looks like you might have a merge conflict with RequestsList ^ @salolivares

**Pull request merged**